terraform/variables.go

```diff
@@ -124,3 +117,38 @@ func mergeInputs(inputsByFile InputsByFile) ValueMap {
 
 	return combinedInputs
 }
+
+var maxLocalsDerefIterations = 32
```

Comment:
This reminds me of the nested directories depth chat. Is there any concern on the number of the chain of locals again? any limitation on the reflect package you're using?
If we think that this is not even a common scenario, why limit ourselves? what do we have to gain here?

Reply:
For example, I prefer getting an error because a user tried to scan something with 23594495 nested locals, than a user having 33 locals and not providing them accurate results.